### PR TITLE
change toChecksumAddress networkId params to 30 and 31. Fix #517

### LINKED ISSUE #517: Regression: tRBTC address checksum is different from that of explorer
**App & Device** (required)
 - Wallet App version: 1.3.1
 - Device: iPhone X
 - OS version: iOS 13.7
 
**Expected behavior**
The testnet tRBTC address should be 
0x05c982D1cD43D37CfEd2E3c2a5512B99C13E9e16 - netID 31 (testnet)
https://explorer.testnet.rsk.co/address/0x05c982d1cd43d37cfed2e3c2a5512b99c13e9e16

While RWallet's is 
0x05C982d1cd43D37CFeD2E3c2a5512b99C13e9e16

### PR DESCRIPTION
**Pull request by AmazingBeerBelly**
CleanShot 2020-10-06 at 15 53 07@2x

Rsk3.utils.toChecksumAddress('0xbD03220C29364b1608F997F2CD454E4177E2E56f', 30);
-> '0xbD03220C29364b1608F997F2CD454E4177E2E56f'

### CODE REVIEW

**Review by chrisli30 — COMMENTED** [review]
@solangegueiros please take a look at this PR fix.

File: src/common/wallet/cointype.js [hunk 1]
```diff
@@ -62,53 +69,60 @@ const coinType = {
...
     icon: references.images.DOC_GREY,
     defaultName: 'Dollar on Chain',
     chain: 'Rootstock',
     type: 'Testnet',
     symbol: 'DOC',
   },
   RIFPTestnet: {
-    networkId: 137,
+    networkId: 31,
+    pathNetworkId: 137,
```
Comment (solangegueiros) [inline src/common/wallet/cointype.js:100]: If it is Testnet, the pathNetworkId is 37310, like the others, isn't it?

File: src/common/wallet/cointype.js [hunk 2]
```diff
@@ -62,53 +69,60 @@ const coinType = {
...
     icon: references.images.RIFP_GREY,
     defaultName: 'RIFPro',
     chain: 'Rootstock',
     type: 'Testnet',
     symbol: 'RIFP',
   },
   RDOCTestnet: {
-    networkId: 137,
+    networkId: 31,
+    pathNetworkId: 137,
```
Comment (solangegueiros) [inline src/common/wallet/cointype.js:109]: If it is Testnet, the pathNetworkId is 37310, like the others, isn't it?

**Review by solangegueiros — COMMENTED** [review]
Thanks for your help, @chrisli30 !
Related to custom tokens it is ok.
I did some comments related to DOCTestnet and RIFPTestnet, they are still using a wrong derivationPath.
I suggest to review, maybe it is not correctly too, I didn't do tests with them.

File: src/common/wallet/cointype.js (hunk 1 above)
Comment (chrisli30) [reply src/common/wallet/cointype.js:100]: Yes, you are right! 👍

**Review by chrisli30 — CHANGES_REQUESTED** [review]
Please correct the pathNetworkId of testnet asset as Sol pointed out.

**Review by chrisli30 — APPROVED** [review]

**Comment by solangegueiros** [comment]
@chrisli30 
I didn't understand where you will use  the pathNetworkId.
It is suppose to be used here: 
https://github.com/rsksmart/rwallet/blob/0795cd943612a58fd43d668de3dbac0af3bf7adc/src/common/wallet/rbtccoin.js#L62

Because the derivationPaths are:
Mainnet: "m/44'/137'/0'/0/"
Testnet:  "m/44'/37310'/0'/0/"

But in the line 62 of rbtccoin.js
`this.path = `m/44'/${this.networkId}'/${this.account}'/0/0`;`
you are still using `networkId`, I think the correct is use `pathNetworkId`.

@bguiz , I asking your help because you are more smart then me to verify if my thoughts are right or not
:)

**Review by solangegueiros — DISMISSED** [review]
Sorry, I block it because my comments

**Comment by chrisli30** [comment]
> you

@solangegueiros 

You absolutely have the rights to request changes if you feel there's something is wrong in the code.

In this case, I see line 62 of rbtc.js has the change. Could you double check again?
`this.path = `m/44'/${this.pathNetworkId}'/${this.account}'/0/0`;`

**Comment by solangegueiros** [comment]
> > you
> 
> @solangegueiros
> 
> You absolutely have the rights to request changes if you feel there's something is wrong in the code.
> 
> In this case, I see line 62 of rbtc.js has the change. Could you double check again?
> 
> image

@chrisli30 , Great! This item is ok now!

But I did another comment...
The `CustomToken` and `CustomTokenTestnet` have the same definition and this is not possible.

image

**Comment by chrisli30** [comment]
> @chrisli30 , Great! This item is ok now!
> 
> But I did another comment...
> The `CustomToken` and `CustomTokenTestnet` have the same definition and this is not possible.
> 
> image
> 
> I found my comment! It is here:
> [0857f94#commitcomment-43039989](https://github.com/rsksmart/rwallet/commit/0857f94d61900e1111d6f0b679fb8949b99c0ebc#commitcomment-43039989)

Yes, @solangegueiros you are right! @AmazingBeerBelly that was a mistake. Please be careful, update it and review the PR carefully!

File: src/common/wallet/cointype.js [hunk 3]
```diff
@@ -13,47 +14,60 @@ const coinType = {
     symbol: 'BTC',
   },
   RBTC: {
-    networkId: 137,
+    networkId: 30,
+    pathNetworkId: 137,
```
Comment (bguiz) [inline src/common/wallet/cointype.js:18]:
```suggestion
    coinType: 137,
```

Let's give this the right name: `coinType`.

This matches the names used in the various standards documents:

- https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
- https://github.com/satoshilabs/slips/blob/master/slip-0044.md
- https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP57.md

(applies to rest of this file too)

File: src/common/wallet/rbtccoin.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ export default class RBTCCoin {
   constructor(symbol, type, path) {
     this.id = type === 'Mainnet' ? symbol : symbol + type;
 
-    // metadata:{network, networkId, icon, defaultName}
+    // metadata:{network, networkId, icon, defaultName, pathNetworkId}
```
Comment (bguiz) [inline src/common/wallet/rbtccoin.js:52]:
```suggestion
    // metadata:{network, networkId, icon, defaultName, coinType}
```

File: src/common/wallet/rbtccoin.js [hunk 5]
```diff
@@ -59,15 +59,15 @@ export default class RBTCCoin {
     // m / purpose' / coin_type' / account' / change / address_index
     this.account = common.parseAccountFromDerivationPath(path);
     this.networkId = this.metadata.networkId;
-    this.path = `m/44'/${this.networkId}'/${this.account}'/0/0`;
+    this.pathNetworkId = this.metadata.pathNetworkId;
+    this.path = `m/44'/${this.pathNetworkId}'/${this.account}'/0/0`;
```
Comment (bguiz) [inline src/common/wallet/rbtccoin.js:63]:
```suggestion
    this.pathNetworkId = this.metadata.cointType;
    this.path = `m/44'/${this.coinType}'/${this.account}'/0/0`;
```

File: src/common/wallet/rbtccoin.js [hunk 6]
```diff
@@ -59,15 +59,15 @@ export default class RBTCCoin {
...
     this.name = this.metadata.defaultName;
-    this.networkId = this.metadata.networkId;
   }
 
   derive(seed) {
     try {
       const master = RBTCCoin.generateMasterFromSeed(seed);
-      const accountNode = RBTCCoin.generateAccountNode(master, this.networkId, this.account);
+      const accountNode = RBTCCoin.generateAccountNode(master, this.pathNetworkId, this.account);
```
Comment (bguiz) [inline src/common/wallet/rbtccoin.js:70]:
```suggestion
      const accountNode = RBTCCoin.generateAccountNode(master, this.coinType, this.account);
```

File: src/common/wallet/rbtccoin.js [hunk 7]
```diff
@@ -107,9 +107,9 @@ export default class RBTCCoin {
     });
   }
 
-  static generateAccountNode(master, networkId, account) {
+  static generateAccountNode(master, pathNetworkId, account) {
```
Comment (bguiz) [inline src/common/wallet/rbtccoin.js:110]:
```suggestion
  static generateAccountNode(master, coinType, account) {
```

File: src/common/wallet/rbtccoin.js [hunk 8]
```diff
@@ -107,9 +107,9 @@ export default class RBTCCoin {
...
     let node = deserializePrivate(master);
-    const path = `m/44'/${networkId}'/${account}'`;
+    const path = `m/44'/${pathNetworkId}'/${account}'`;
```
Comment (bguiz) [inline src/common/wallet/rbtccoin.js:112]:
```suggestion
    const path = `m/44'/${coinType}'/${account}'`;
```

**Review by bguiz — CHANGES_REQUESTED** [review]
Also need to check rest of code base where `networkId` is used.

**Review by solangegueiros — COMMENTED** [review]
It seems ok now.

@chrisli30 , @bguiz
Question:
What will happen with the wallets which exists before, until now and was wrong?

They will still work?
I think the people will lost the funds which was in the old address. am I right?
The people will be able to recover / transfer your funds?
Do you have a plan for this?

**Comment by chrisli30** [comment]
> It seems ok now.
> 
> @chrisli30 , @alebanzas
> Question:
> What will happen with the wallets which exists before, until now and was wrong?
> 
> They will still work?
> I think the people will lost the funds which was in the old address. am I right?
> The people will be able to recover / transfer your funds?
> Do you have a plan for this?

If you look at the old code,
`this.path = `m/44'/${this.networkId}'/${this.account}'/0/0`;`

The {networkId} is either 137 for mainnet or 37310 for testnet. When we merge this fix and create a build, wallet addresses will have no change.

File: src/common/wallet/cointype.js (hunk 3 above)
Comment (chrisli30) [reply src/common/wallet/cointype.js:18]: I see. That makes sense. Okay we will do another sweep and rename the pathNetworkId to `coinType`. Yes other files need to be checked too.

**Review by chrisli30 — APPROVED** [review]

**Review by imstar15 — APPROVED** [review]

**Review by bguiz — APPROVED** [review]

**Review by solangegueiros — APPROVED** [review]
Sorry, I travelled and I can't imagine that I do not have internet here.
A approved, but I still have doubts about some address, but we can check it in another PR.

**Pull request merged**